lib/sensors/stripe_sensor.cpp

```diff
@@ -0,0 +1,21 @@
+#include "stripe_sensor.hpp"
+
+namespace hyped::sensors {
+
+    Sensor::Sensor(const std::uint8_t newPin){
+        pin = newPin;
+        HardwareGpio sensor(&log_);
+        sensor.getReader(newPin);
+    };

```

Comment:
`pin`, initialise class values in the initialiser list 
ie 
`Sensor::Sensor(const std::uint8_t pin):pin_(pin), logger_(logger) {`
`    // do stuff`
`}`

also you need to include `log` in the constructor args, plus should be called `logger_` for consistency across the repo

Reply:
also this should not be class `Sensor`, probably `Keyence` but I'm not sure what the official term for it is

Reply:
and the reader will need to be stored in a class data member